app/views/admins/_user_entry.haml

```diff
@@ -24,6 +24,9 @@
         - unless user.person.closed_account
           %li= link_to t('admins.user_search.close_account'), admin_close_account_path(user), method: :post, data: { confirm: t('admins.user_search.are_you_sure') }, class: 'btn btn-danger btn-mini'
 
+        - unless user.locked_at
+          %li= link_to t('admins.user_search.lock_account'), admin_close_account_path(user), method: :post, data: { confirm: t('admins.user_search.are_you_sure_lock_account') }, class: 'btn btn-danger btn-mini'
```

Comment:
You should be able to use https://github.com/plataformatec/devise/blob/master/lib/devise/models/lockable.rb#L59